### ISSUE TITLE
optimizer 2:1 or 1:1

### ISSUE THREAD

**Issue by Aikawa24** [ProudElm/packaging_solaredgeoptimizers#12]
hi, my P850 Optimisers are connected with 2 PV Modules, the SE monitoring shows it correctly, is there a way to get this 2:1 or 1:1 connection status as attribute?

**Comment by ProudElm** [comment]
You layout is different than any i have seen. The code does not work correcly for this. Where the code expacts a string (containing 1 or more optimizers) your json has optimizers. Thus the code does not works.

**Comment by Aikawa24** [comment]
The Layout is one string for east and one string for west,
A Se16K with two strings.
String west has 27 PV Modules where two Modules are connected serial to one p850 optimizer and the last pv module is connected alone to the p850.
The same on the east String.

The code works but it shows not the module energy buu the energy value

Am 08/07/2023 um 15:12 schrieb ProudElm ***@***.***>:

﻿

You layout is different than any i have seen. The code does not work correcly for this. Where the code expacts a string (containing 1 or more optimizers) your json has optimizers. Thus the code does not works.

**Comment by ProudElm** [comment]
i see.. never thought about 2 panels per 1 optimizer. The code does not find panels but the optimizers instead. So when there are more then 1 panel per optimizer this code fails retrieving all the data.

In your case it only finds the 15 optimizers on string 1 and the 14 on string 2. It will now also report the values for those (i think). But that is indeed not wat you would want.

I'm trying to figure out if i can find a workaround.

**Comment by ProudElm** [comment]
I now have something working with will check if the optimizer has a difference in the fields energy and moduleenergy we have a optimizer with more then 1 panel (for now, i assume 2). And i will add the same optimizer again.
This will result in a list of all optimizers (some dubbel thus).

The problem now is getting the data. Hopefully it will work soon!

**Comment by ProudElm** [comment]
i almost have it working, but i'm stuck at the point that i do not know what the response is from the request i make to retrieve the power/voltage etc value's. Are those also just doubled? 
 
Can you perhaps send me a part of the result from this request (Should work in a browser if you are also logging in the portal):

https://monitoringpublic.solaredge.com/solaredge-web/p/publicSystemData?reporterId=125193141&type=panel&activeTab=0&fieldId=<<SITEID>>&isPublic=true&locale=en_US

(the number: 125193141 is one of your optimizers, 1 optimizer, 2 panels).

Im only interessed in the firt part: the SE.systemData. Thank you!

**Comment by Aikawa24** [comment]
Hi, the Voltage is doubled.

`SE.systemData = {"countrySettings":{},"description":"Panel 1.2.9","lastMeasurementDate":"Thu Jul 20 09:03:39 GMT 2023","manufacturer":"Hanwha Q.Cells GmbH","measurements":{"Current [A]":"1.36","Optimizer Voltage [V]":"53.75","Power [W]":"90.02","Voltage [V]":"66.38"},"model":"Q.PEAK DUO-G8 345","phase1Measurements":{},"phase2Measurements":{},"phase3Measurements":{},"secondarySerialNumbers":{"serialNumber":"119A8648-79"},"serialNumber":"119A8648-79","softwareInformation":{},"status":"Active","type":"POWER_BOX"};
SE.secondarySerialNumbers = {"serialNumber":"119A8648-79"};
SE.secondarySerialNumbersKeySet = '[serialNumber]';`

**Comment by ProudElm** [comment]
thanx!

So that optimizer is doing now 90W but this means for both panels it's 45W each? Or are both panels doing 90W? I would assume all those values are doubled, but im not sure if that is indeed the case.
Assuming that the Voltage is the one that counts and not the optimizer voltage the values are all doubled or none.
P=U*I. 1.36A*66.38V = 90,28W.  Close enough.

this is becoming a bit of a weird issue. I always considered optimizers and panels the same. But here we have a case where that is not true. Im using optimizer in all places as name etc.. We could also say, we just report optimizer data only and then we dont care if one 1 or more panels are attteched to each optimizer. (the easy way). Or we switch to panels and change all the names etc.. Wich is a lot of work and could break things (the hard way).
I will think on this a bit more.

**Comment by Aikawa24** [comment]
the power is doubled too, that was the problem, I have to split the Value for each Optimizer in HA and make a new single Value for my Panels to show the Layout With power.

another problem is the Values are not time equal, some optimiser Values has Timestamps e.g. 09:23:30 and others 09:24:45 or 09:22:15 so if I want to see which Panel makes the best performance "now" there is a big time difference between their Values.

i don't know how these difference occur, is it because your script loads it slowly or the optimisers pushed not synchrony on the same Time

**Comment by ProudElm** [comment]
Yes, and i can fix that somehow in the code, by adding an extra optimizer for each optimizer who has 2 panels. I will spil the values for those optimizers.

About the time, there is nothing i can do about that. Not sure why the portal is limiting that or that the inverter is to blame for not transmitting the data at the same time. But i cant do anything about that.

Best we can do is hope one day all this has become absolute and al the data is retrieved direct via modbos from the inverter.
